canary/producer-cpp/src/Canary.cpp

```diff
@@ -666,15 +351,24 @@ int gstreamer_test_source_init(CustomData *data, GstElement *pipeline) {
     autovidcon = gst_element_factory_make("autovideoconvert", "vidconv");
     h264enc = gst_element_factory_make("x264enc", "h264enc");
     h264parse = gst_element_factory_make("h264parse", "h264parse");
-    appsink = gst_element_factory_make("appsink", "appsink");
+    kvssink = gst_element_factory_make("kvssink", "kvssink");
     h264parse = gst_element_factory_make("h264parse", "h264parse");
 
     // videotestsrc must be set to "live" in order for pts and dts to be incremented
     g_object_set(source, "is-live", TRUE, NULL);
 
-    // configure appsink
-    g_object_set(G_OBJECT (appsink), "emit-signals", TRUE, "sync", FALSE, NULL);
-    g_signal_connect(appsink, "new-sample", G_CALLBACK(on_new_sample), data);
+    // configure kvssink
+    g_object_set(G_OBJECT (kvssink), "stream-name", data->streamName, "storage-size", 128, NULL);
+
+    string use_iot_cred(data->pCanaryConfig->use_iot_credential_provider);
+    transform(use_iot_cred.begin(), use_iot_cred.end(),use_iot_cred.begin(), ::tolower);
+
+    if(use_iot_cred.compare("true") == 0){
```

Comment:
This block seems weird. determine should be invoked either ways. Basically, the check for iot or not should happen within `determine_credentials`. If iot and if the required isnt set, the job should fail. If it is not IoT, access and secret should be set and if it isnt set, we need to fail. In either cases, we should push a status code to the cloudwatch (preferred) or log and fail the job.